cabal-install/src/Distribution/Client/InstallSymlink.hs

```diff
@@ -195,12 +197,20 @@ symlinkBinary overwritePolicy publicBindir privateBindir publicName privateName
     OkToOverwrite     -> rmLink >> mkLink >> return True
     NotOurFile ->
       case overwritePolicy of
-        NeverOverwrite  ->                     return False
-        AlwaysOverwrite -> rmLink >> mkLink >> return True
+        NeverOverwrite  -> return False
+        AlwaysOverwrite -> overwrite
+        PromptOverwrite -> maybeOverwrite
   where
     relativeBindir = makeRelative publicBindir privateBindir
     mkLink = createFileLink (relativeBindir </> privateName) (publicBindir   </> publicName)
     rmLink = removeFile (publicBindir </> publicName)
+    overwrite = True <$ (rmLink *> mkLink)
+    maybeOverwrite :: IO Bool
+    maybeOverwrite = do
+      a <- promptYesNo
+        "Existing file found while installing symlink. Do you want to unlink that file? (y/n)"
```

Comment:
From a user perspective the phrasing of the second sentence is a bit obscure. Maybe a better phrasing would be "Do you want to overwrite that file?"

Reply:
Accepted